runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -671,10 +689,14 @@ private void checkClusterId(@Nonnull Layout layout) {
      * This future will continue retrying until it gets a layout.
      * If you need this completable future to fail, you should chain it with a timeout.
      *
+     * @param layoutServers Layout servers to fetch the layout from.
      * @return A completable future containing a layout.
      */
-    private CompletableFuture<Layout> fetchLayout() {
-        return CompletableFuture.<Layout>supplyAsync(() -> {
+    private CompletableFuture<Layout> fetchLayout(List<String> layoutServers) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 22 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)